src/pokemanki/gui/pokemanki_trade.py

```diff
@@ -0,0 +1,175 @@
+
+from .forms.qt6 import pokemanki_trade
+
+
+class TradeWindow(QDialog):
+    def __init__(self, parentObj, mw):
+        super().__init__()
+        self.dialog = pokemanki_trade.Ui_Dialog()
+        self.dialog.setupUi(self)
+        self.setupUi(parentObj, mw)
+
+    def setupUi(self, parent, mw):
+        self.dialog.webEngineView = AnkiWebView(parent=self, title="pokémanki trades")
+        sizePolicy = QtWidgets.QSizePolicy(
+            QtWidgets.QSizePolicy.Policy.Expanding,
+            QtWidgets.QSizePolicy.Policy.Expanding,
+        )
+        sizePolicy.setHorizontalStretch(0)
+        sizePolicy.setVerticalStretch(0)
+        sizePolicy.setHeightForWidth(
+            self.dialog.webEngineView.sizePolicy().hasHeightForWidth()
+        )
+        self.dialog.webEngineView.setSizePolicy(sizePolicy)
+        self.dialog.webEngineView.setObjectName("webEngineView")
+        self.dialog.verticalLayout_2.addWidget(self.dialog.webEngineView)
+
+        self.dialog.webEngineView.set_bridge_command(parent.on_bridge_cmd, parent)
+
+        # Set in the middle of the screen
+        width = 800
+        height = self.height() + 175
+
+        if platform.system() == "Windows":
+            user32 = ctypes.windll.user32
+            posX = int(user32.GetSystemMetrics(0) / 2 - width / 2)
+            posY = int(user32.GetSystemMetrics(1) / 2 - height / 2)
+        else:
+            posX = int(mw.frameGeometry().width() / 2 - width / 2)
+            posY = int(mw.frameGeometry().height() / 2 - height / 2)
+
+        self.setGeometry(posX, posY, width, height)
+
+    def setup_trades(self, trades):
+        """Set up the web view's html.
+
+        :param list trades: List of trades to display
+        """
+
+        self.dialog.webEngineView.stdHtml(
+            body=_trades_html(trades),
+            css=["/pokemanki_css/view_trade.css", "/pokemanki_css/main.css"],
+            context=self,
+        )
+
+    def open_trades(self, trades):
+        """Set up the web view's html and opens the trade dialog.
+
+        :param list trades: List of trades to display
+        """
+
+        self.setup_trades(trades)
+        self.open()
+
+    def finished_trade(self):
+        self.done(QDialog.DialogCode.Accepted)
+
+
+def _trades_html(trades):
+    """Generate the html code for the trades window.
+
+    :param list trades: List of trades to display
+    :return: The html code.
+    :rtype: str
+    """
+    # Open trades container
+    txt = '<div class="pk-td-container">'
+
+    # Generate each of the trades
+    for i in range(len(trades)):
+        txt += _trade_html(i, trades)
+
+    # Close trades container
+    txt += "</div>"
+
+    return txt
+
+
+def _trade_html(i, trades):
+    """Generate the html code for a trade.
+
+    :param int i: Trade number.
+    :param list trades: List of trades to display
+    :return: Trade's html.
+    :rtype: str
+    """
+
+    trade = "<script>" "function callTrade(n) { pycmd(n); }" "</script>"
```

Comment:
Nit: This function only needs to be defined once, regardless of how many trades are available. Best move this line to `_trades_html`.